contrib/p4/docker_files/ptf_tester/tests/test_no_fail.py

```diff
@@ -0,0 +1,172 @@
+    def __init__(self):
+        DataplaneBaseTest.__init__(self)
+
+    def runTest(self):
+        # log_to_file("Staring first test")
+        pkt = testutils.simple_tcp_packet(
+            pktlen=100,
+            ip_dst="192.168.1.3",
+            ip_ttl=64,
+            eth_dst="00:01:02:03:04:05",
+            eth_src="00:01:02:03:04:05",
+        )
+        pkt_expected = testutils.simple_tcp_packet(
+            pktlen=100,
+            ip_dst="192.168.1.3",
+            ip_ttl=63,
+            eth_dst="00:01:02:03:04:05",
+            eth_src="00:01:02:03:04:05",
+        )
+        try:
+            testutils.send_packet(self, (1, 1), pkt)
+        except Exception as e:
+            print(e)
+
+        testutils.verify_packet(self, pkt_expected, (2, 0))
+
+
+class Ipv4TwoSwitchesTest(DataplaneBaseTest):
+    def __init__(self):
+        DataplaneBaseTest.__init__(self)
+
+    def runTest(self):
+        pkt = testutils.simple_tcp_packet(
+            pktlen=100,
+            ip_dst="192.168.1.4",
+            ip_ttl=64,
+            eth_dst="00:01:02:03:04:05",
+            eth_src="00:01:02:03:04:05",
+        )
+        pkt_expected = testutils.simple_tcp_packet(
+            pktlen=100,
+            ip_dst="192.168.1.4",
+            ip_ttl=62,
+            eth_dst="00:01:02:03:04:05",
+            eth_src="00:01:02:03:04:05",
+        )
+
+        testutils.send_packet(self, (0, 0), pkt)
+
+        testutils.verify_packet(self, pkt_expected, (3, 0))
+
+
+class LongChainOfSwitches(DataplaneBaseTest):
+    def __init__(self):
+        DataplaneBaseTest.__init__(self)
+
+    def runTest(self):
+        pkt = testutils.simple_tcp_packet(
+            pktlen=100,
+            ip_dst="192.168.1.16",
+            ip_ttl=64,
+            eth_dst="00:01:02:03:04:05",
+            eth_src="00:01:02:03:04:05",
+        )
+
+        pkt_expected = testutils.simple_tcp_packet(
+            pktlen=100,
+            ip_dst="192.168.1.16",
+            ip_ttl=58,
+            eth_dst="00:01:02:03:04:05",
+            eth_src="00:01:02:03:04:05",
+        )
+
+        testutils.send_packet(self, (0, 0), pkt)
+
+        testutils.verify_packet(self, pkt_expected, (15, 0))
+
+
+# class MyTunnelPacketTest(DataplaneBaseTest):
+#     def __init__(self):
+#         DataplaneBaseTest.__init__(self)
+
+#     def runTest(self):
+#         #log_to_file("Starting second test")
+
+#         bind_layers(Ether, MyTunnel, type=TYPE_MYTUNNEL)
+#         bind_layers(MyTunnel, IP, pid=TYPE_IPV4)
+#         pkt = Ether(dst="00:01:02:03:04:05", src="00:01:02:03:04:05")/MyTunnel()/IP(dst="192.168.1.2", ttl=64)
+
+#         log_to_file("---Packet--- \n" + str(pkt.show))
+
+#         pkt_expected = Ether(dst="00:01:02:03:04:05", src="00:01:02:03:04:05")/IP(dst="192.168.1.2", ttl=63)
+
+#         testutils.send_packet(self, (0,1), pkt)
+
+#         testutils.verify_packet(self, pkt_expected, (1, 1))
+
+
+def log_to_file(msg):
```

Comment:
Seems unused (but if used, would better be rewritten or replaced by regular logging statements?).